receiver/oktareceiver/config.go

```diff
@@ -0,0 +1,102 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oktareceiver // import "github.com/observiq/bindplane-agent/receiver/oktareceiver"
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+)
+
+var (
+	defaultPollInterval = time.Minute
+
+	// ISO 8601 Format
+	OktaTimeFormat = "2006-01-02T15:04:05Z"
+)
+
+// Config defines the configuration for an Okta receiver
+type Config struct {
+	// Domain Okta Domain (no https://  -  ex: observiq.okta.com)
+	Domain string `mapstructure:"okta_domain"`
+
+	// ApiToken Okta Api Token
+	ApiToken string `mapstructure:"api_token"`
+
+	// PollInterval The interval at which the Okta API is scanned for Logs
+	// Must be 1s or greater
+	PollInterval time.Duration `mapstructure:"poll_interval"`
+
+	// StartTime UTC Timestamp following format specified in OktaTimeFormat
+	// Must be within the past 180 days and not in the future
+	StartTime string `mapstructure:"start_time"`
+}
+
+var (
+	errNoDomain            = errors.New("okta_domain must be specified")
+	errInvalidDomain       = errors.New("invalid okta_domain, do not include https://")
+	errNoApiToken          = errors.New("api_token must be specified")
+	errInvalidPollInterval = errors.New("invalid poll_interval, it must be a duration greater than one second")
+)
+
+// Validate ensures an Okta receiver config is correct
+func (c *Config) Validate() error {
+	if c.Domain == "" {
+		return errNoDomain
+	}
+
+	if strings.Contains(c.Domain, "https://") || strings.Contains(c.Domain, "http://") {
+		return errInvalidDomain
+	}
+
+	if c.ApiToken == "" {
+		return errNoApiToken
+	}
+
+	if c.PollInterval != 0 && c.PollInterval < time.Second {
```

Comment:
I hadn't yet learned how bindplane sends the default value for optional pararmeters, and thought it would be sending over the zero value of the field. But I see now that it has a prepopulated default value that I have now set to "1m". I'll remove this case.